### PR TITLE
Fix identity select statement caching

### PR DESCRIPTION
**Pull request by beikov**

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/persister/entity/AbstractEntityPersister.java [hunk 1]
```diff
@@ -833,15 +833,15 @@ else if ( batchSize > 1 ) {
 	}
 
 	private String getIdentitySelectString(Dialect dialect) {
-		if ( dialect.getIdentityColumnSupport().supportsInsertSelectIdentity() ) {
```
Comment (gavinking) [inline hibernate-core/src/main/java/org/hibernate/persister/entity/AbstractEntityPersister.java:836]: Really, what was wrong with that?

Comment (gavinking) [reply hibernate-core/src/main/java/org/hibernate/persister/entity/AbstractEntityPersister.java:836]: Oh I see. Yeah, it's wrong.

Comment (beikov) [reply hibernate-core/src/main/java/org/hibernate/persister/entity/AbstractEntityPersister.java:836]: See https://ci.hibernate.org/job/hibernate-orm-nightly/job/main/78/testReport/junit/org.hibernate.orm.test.id/IdentityIdEntityTest/Build___h2_1_4___Test___testIdentityEntityWithDisabledGetGeneratedKeys_SessionFactoryScope_/

Comment (gavinking) [reply hibernate-core/src/main/java/org/hibernate/persister/entity/AbstractEntityPersister.java:836]: Oops. Nice catch.

**Pull request merged**